src/internal/_xtake.js

```diff
@@ -15,7 +15,8 @@ module.exports = (function() {
       return _reduced(result);
     } else {
       this.n -= 1;
-      return this.xf['@@transducer/step'](result, input);
+      return this.n === 0 ? _reduced(this.xf['@@transducer/step'](result, input))
+                          : this.xf['@@transducer/step'](result, input);
     }
```

Comment:
What else is going on after `this.xf['@@transducer/step'](result, input))` is returned?
Maybe we can achieve the desired behaviour somewhere else, I'm just having a hard time tracing the calls. I don't get why calling `_reduce` on the result of the call, vs on the result on the next iteration would have different behaviour.